src/utils/spo.ts

```diff
@@ -1485,6 +1488,122 @@ export const spo = {
     await request.post(requestOptions);
   },
 
+  async getFileAsListItemByUrl(webUrl: string, url: string, logger?: Logger, verbose?: boolean): Promise<any> {
+    if (verbose && logger) {
+      logger.logToStderr(`Getting the file properties with url ${url}`);
+    }
+
+    const serverRelativePath = urlUtil.getServerRelativePath(webUrl, url);
+    const requestUrl = `${webUrl}/_api/web/GetFileByServerRelativePath(DecodedUrl=@f)?$expand=ListItemAllFields&@f='${formatting.encodeQueryParameter(serverRelativePath)}'`;
+
+    const requestOptions: CliRequestOptions = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const file = await request.get<FileProperties>(requestOptions);
+
+    return file.ListItemAllFields;
+  },
+
+  /**
+  * Updates a list item with system update
+  * @param webUrl Web url
+  * @param requestUrl The base URL without query parameters, pointing to the specific list where the item resides. This URL should represent the list.
+  * @param id The id of the list item
+  * @param properties An object of the properties that should be updated
+  * @param contentTypeName The name of the content type to update
+  * @param logger The logger object
+  * @param verbose If in verbose mode
+  * @returns The updated list item object
+  */
+  async systemUpdateListItem(requestUrl: string, id: string, logger: Logger, verbose: boolean, properties?: object, contentTypeName?: string): Promise<ListItemInstance> {
+    if (verbose && logger) {
+      logger.logToStderr(`getting request digest for systemUpdate request`);
+    }
+
+    const parsedUrl = new URL(requestUrl);
+    const serverRelativeSiteUrl = requestUrl.match(new RegExp('/sites/[A-Za-z0-9-]*'))![0];
```

Comment:
Isn't `_` an allowed character? Rather than making our own list of allowed character, wouldn't it be safer to say: everything except a slash which indicates the next URL segment (`[^/]+`)?